lib/components/admin/query-record.js

```diff
@@ -1,37 +1,51 @@
+import { planParamsToQuery } from '@opentripplanner/core-utils/lib/query'
+import { getTimeFormat, OTP_API_TIME_FORMAT } from '@opentripplanner/core-utils/lib/time'
+import moment from 'moment'
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
 
 import * as formActions from '../../actions/form'
-import Icon from '../narrative/icon'
+import {CallRecordButton, CallRecordIcon} from './styled'
 
 /**
  * Displays information for a query stored for the Call Taker module.
  */
 class QueryRecordLayout extends Component {
+  _getParams = () => {
+    const {query} = this.props
+    return planParamsToQuery(JSON.parse(query.queryParams))
+  }
+
   _viewQuery = () => {
-    const {parseUrlQueryString, query} = this.props
-    const params = JSON.parse(query.queryParams)
-    if ('arriveBy' in params) {
-      params.departArrive = params.arriveBy ? 'ARRIVE' : 'DEPART'
-    }
-    parseUrlQueryString(params)
+    const {parseUrlQueryString} = this.props
+    const params = this._getParams()
+    params.departArrive = params.arriveBy ? 'ARRIVE' : 'DEPART'
+    parseUrlQueryString(params, '_CALL')
   }
 
   render () {
-    const {index} = this.props
+    const {query, timeFormat} = this.props
+    const params = this._getParams()
+    const time = query.timeStamp
+      ? moment(query.timeStamp).format(timeFormat)
+      : moment(params.time, OTP_API_TIME_FORMAT).format(timeFormat)
     return (
       <li>
-        <button onClick={this._viewQuery} className='clear-button-formatting'>
-          <Icon type='search' />
-        </button>{' '}
-        Query {index + 1}
+        <CallRecordButton onClick={this._viewQuery} className='clear-button-formatting'>
+          <CallRecordIcon type='search' />
+          {time}<br />
+          {params.from.name} to {params.to.name}
+        </CallRecordButton>
       </li>
     )
   }
 }
 
 const mapStateToProps = (state, ownProps) => {
-  return {}
+  return {
+    config: state.otp.config,
```

Comment:
Is this needed?

Reply:
removed in 9527a56db41e2a7cb9e28ae553a669b4fc206a2b